modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -67,6 +59,24 @@ object BlockService {
 
         def getCollateral: Amount = collateral
       }
+
+      private def processAcceptanceSuccess(hashedBlock: Hashed[DAGBlock])(contextUpdate: BlockAcceptanceContextUpdate) = for {
+        hashedTransactions <- hashedBlock.signed.transactions.toNonEmptyList
+          .sortBy(_.ordinal)
+          .traverse(_.toHashed)
+
+        _ <- hashedTransactions.traverse(transactionStorage.accept)
+        _ <- blockStorage.accept(hashedBlock)
+        _ <- addressStorage.updateBalances(contextUpdate.balances)
+        isDependent = (block: Signed[DAGBlock]) => BlockRelations.dependsOn(Set(hashedBlock), Set.empty)(block)
```

Comment:
I see couple of possible improvements there:

1. You can make use of the overload which you wrote (https://github.com/Constellation-Labs/tessellation/pull/539/files#diff-bae38fdffd01e17ed0cbee7f326d32021df5926f82f2c87d0e72a89d821be34aR18) which already lifts `hashedBlock` into `Set` and makes use of `Set.empty` as empty param:
```scala
isDependent = (block: Signed[DAGBlock]) => BlockRelations.dependsOn(hashedBlock)(block)
```

2. Double curring is redundant because function is already curried

```scala
isDependent = BlockRelations.dependsOn(hashedBlock) // it returns (block: Signed[DAGBlock]) => F[Boolean]
```

3. Having previous points in mind I would just use it directly:

```scala
_ <- blockStorage.restoreDependent(BlockRelations.dependsOn(hashedBlock))
```

Reply:
Second point is invalid as there are also implicits. Yet, @kriscieslak you can check the last one

Reply:
It doesn't work because of implicits as well.